sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosWriterBase.scala

```diff
@@ -91,27 +95,69 @@ private abstract class CosmosWriterBase(
     }
 
     val partitionKeyValue = PartitionKeyHelper.getPartitionKeyPath(objectNode, partitionKeyDefinition)
-    writer.scheduleWrite(partitionKeyValue, objectNode)
+    writer.get.scheduleWrite(partitionKeyValue, objectNode)
   }
 
   override def commit(): WriterCommitMessage = {
     log.logInfo("commit invoked!!!")
-    writer.flushAndClose()
+
+    try {
+      writer.get.flushAndClose()
+    } catch {
+      case bulkWriterStaleError: BulkWriterNoProgressException =>
+        bulkWriterStaleError.activeBulkWriteOperations match {
+          case Some(remainingWriteOperations) =>
+            log.logWarning(s"Error indicating stuck writer when committing writes. Retry will be attempted for "
+              + s"the outstanding ${remainingWriteOperations.size} write operations.", bulkWriterStaleError)
+
+            val bulkWriterForRetry =
+                new BulkWriter(
+                  container,
+                  partitionKeyDefinition,
+                  cosmosWriteConfig,
+                  diagnosticsConfig,
+                  getOutputMetricsPublisher(),
+                  commitAttempt.getAndIncrement())
```

Comment:
should here be incrementAndGet?

Reply:
synced offline, need to pass the original value

Reply:
No - I intentionally start with 1 - so, getAndIncrement is intentional